lib/price-import.coffee

```diff
@@ -134,6 +137,15 @@ class PriceImport extends ProductImport
     debug 'About to send %s requests', _.size(posts)
     Promise.settle(posts)
 
+  ###
+   * returns true if product does not have any staged changes
+   * and is published
+  ###
+  _canBePublished: (product) ->
+    if !product.hasStagedChanges and product.published
```

Comment:
This is a nice idea to make it more generic. We can have it as a separate issue and work on it as an enhancement to this feature.
But currently so much generic feature was not required.